server/src/main/java/io/crate/auth/HttpAuthUpstreamHandler.java

```diff
@@ -119,7 +120,8 @@ private void handleHttpRequest(ChannelHandlerContext ctx, HttpRequest request) {
         }
 
         InetAddress address = addressFromRequestOrChannel(request, ctx.channel());
-        ConnectionProperties connectionProperties = new ConnectionProperties(credentials, address, Protocol.HTTP, session);
+        Duration maxAge = cacheControlMaxAgeFromRequest(request);
+        ConnectionProperties connectionProperties = new ConnectionProperties(credentials, address, Protocol.HTTP, session, maxAge);
```

Comment:
We can't use the cache-control header from the request that's being made to CrateDB but instead we need to use the cache-control header of the response from the request we make to the jwk endpoint.

On a glance over the `UrlJwkProvider` implementation it doesn't look that complex and we could implement our on version that includes per domain caching.